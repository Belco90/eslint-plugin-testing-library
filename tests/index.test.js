'use strict';

const { configs, rules } = require('../lib');
const fs = require('fs');
const path = require('path');

const rulesModules = fs.readdirSync(path.join(__dirname, '/lib/rules'));

it('should export proper rules', () => {
  const availableRules = rulesModules.map(module => module.replace('.js', ''));
  expect(Object.keys(rules)).toEqual(availableRules);
});

it.each(['recommended', 'angular', 'react', 'vue'])(
  'should export proper "%s" config',
  configName => {
    expect(configs[configName]).toMatchSnapshot();
  }
);
