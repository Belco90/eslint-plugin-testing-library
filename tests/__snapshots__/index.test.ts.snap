// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should export proper "angular" config 1`] = `
Object {
  "plugins": Array [
    "testing-library",
  ],
  "rules": Object {
    "testing-library/await-async-query": "error",
    "testing-library/await-async-utils": "error",
    "testing-library/no-await-sync-query": "error",
    "testing-library/no-container": "error",
    "testing-library/no-debug": "warn",
    "testing-library/no-dom-import": Array [
      "error",
      "angular",
    ],
    "testing-library/no-node-access": "error",
    "testing-library/no-promise-in-fire-event": "error",
    "testing-library/no-wait-for-empty-callback": "error",
    "testing-library/prefer-find-by": "error",
    "testing-library/prefer-screen-queries": "error",
    "testing-library/prefer-user-event": "warn",
  },
}
`;

exports[`should export proper "dom" config 1`] = `
Object {
  "plugins": Array [
    "testing-library",
  ],
  "rules": Object {
    "testing-library/await-async-query": "error",
    "testing-library/await-async-utils": "error",
    "testing-library/no-await-sync-query": "error",
    "testing-library/no-promise-in-fire-event": "error",
    "testing-library/no-wait-for-empty-callback": "error",
    "testing-library/prefer-find-by": "error",
    "testing-library/prefer-screen-queries": "error",
    "testing-library/prefer-user-event": "warn",
  },
}
`;

exports[`should export proper "react" config 1`] = `
Object {
  "plugins": Array [
    "testing-library",
  ],
  "rules": Object {
    "testing-library/await-async-query": "error",
    "testing-library/await-async-utils": "error",
    "testing-library/no-await-sync-query": "error",
    "testing-library/no-container": "error",
    "testing-library/no-debug": "warn",
    "testing-library/no-dom-import": Array [
      "error",
      "react",
    ],
    "testing-library/no-node-access": "error",
    "testing-library/no-promise-in-fire-event": "error",
    "testing-library/no-wait-for-empty-callback": "error",
    "testing-library/prefer-find-by": "error",
    "testing-library/prefer-screen-queries": "error",
    "testing-library/prefer-user-event": "warn",
  },
}
`;

exports[`should export proper "vue" config 1`] = `
Object {
  "plugins": Array [
    "testing-library",
  ],
  "rules": Object {
    "testing-library/await-async-query": "error",
    "testing-library/await-async-utils": "error",
    "testing-library/await-fire-event": "error",
    "testing-library/no-await-sync-query": "error",
    "testing-library/no-container": "error",
    "testing-library/no-debug": "warn",
    "testing-library/no-dom-import": Array [
      "error",
      "vue",
    ],
    "testing-library/no-node-access": "error",
    "testing-library/no-promise-in-fire-event": "error",
    "testing-library/no-wait-for-empty-callback": "error",
    "testing-library/prefer-find-by": "error",
    "testing-library/prefer-screen-queries": "error",
    "testing-library/prefer-user-event": "warn",
  },
}
`;
